File: Programming Languages/assignment1/InsertionSort.cpp
#include <sstream>
#include <iostream>
using namespace std;
//craete InsertionSort function
void InsertionSort(){
	int length;
	cout << "Hello, and welcome to sorting 101" << endl;
	cout << "Please enter in a length for how many numbers you wish to sort, without letters, spaces, or special characters." << endl;
	cin >> length;
	//length = length-1;
	if(!cin){
		cin.clear();
		cin >> length;
	}
	//create data array and set up variables for swapping
	int data [length];
	//ask for the data
	cout << "Now please enter in the numbers you wish to organize, seperated by spaces. Warning: if the list is out of bounds, data will be lost." << endl;
	cout << "Enter in the numbers set in the format '1' enter, '2' enter, for each element." << endl;
	int text;
	for(int i=0; i<length; i++){
		cin >> text;
		data[i] = text;
	}
	//print out the unsorted data
	cout << "unsorted List: " << endl;
	for(int i=0; i<length-1; i++){
		//add formatting
		cout<< " "<< data[i] << ",";
	}
	cout<< ""<<endl;
	//create sorting function portion of code
	//create 'j' and 'temp' incase of position switching.
	int j=0;
	int temp;
	for (int i= 1; i < length; i++){
		//going through each element...
		//set "temp" to the current element
		temp= data[i];
		//go through all elements before, and if less than current element, swap current element.
		for (j= i-1; (j>=0) && (temp<data[j]); j--){
			data[j+1] = data[j];
		}
		data[j+1]= temp;
	}
	//now the data is sorted, print out results
	cout << "Sorted list: "<<endl;
	for(int p=0; p<length-1; p++){
		//add in formatting for the data
		cout << data[p] << ", ";
	}
}
//run InsertionSort
int main(){

	InsertionSort();
}

File: Programming Languages/assignment1/InsertionSort.java
//InsertionSort for Java by Joshua Ewart
import java.util.ArrayList;
import java.util.*;
//create class to contain project
public class InsertionSort {
	//create an array to sort the data
	public static ArrayList<Integer> data = new ArrayList<Integer>();
	//create a method to add in the elements
	public static void addData(int input){
		data.add(input);
	}
	//create method to run insertionsort
	public static void insertionSort(){
		//create variables for going through the data
		int temp, i, j;
		//go through each element of the array
		for(i=1;i<data.size();i++){
			//set temp variable 'j' to current position
			temp= data.get(i);
			//go through until reach the begining, and the too values are not equal
			for(j=i-1; j>=0 && temp<data.get(j); j--){
				//swap the variables of elements, if data[i] is less than than data[j]
				data.set(j+1, data.get(j));
			}
			//then set the next element back to temp
			data.set(j+1, temp);
		}
	}
	//create main method to run the program
	public static void main(String[] args){
		//add in the data
		Scanner input = new Scanner(System.in);
		System.out.println("Please enter in the number of elements");
		int length = input.nextInt();
		System.out.println("Please enter in each element, as an integer, followed by pressing 'enter'.");
		for(int i=0; i<length; i++){
			System.out.print("Please enter in element: ");
			System.out.println(i+1);
			int element = input.nextInt();
			addData(element);
		}
		//display unsorted data:
		System.out.println("Unsorted: ");
		//go through each element
		for(int i=0; i<length; i++){
			System.out.print(data.get(i));
			//create formatting
			if(i != 9){
				System.out.print(", ");
			}
		}
		System.out.println("");
		//run sorting algorithm method
		insertionSort();		
		//Print Sorted data
		System.out.print("Sorted: ");
		//itterate through each element of the array
		for(int i=0; i<10; i++){
			System.out.print(data.get(i));
			//create formatting for the output
			if(i != 9){
				System.out.print(", ");
			}	
		}
	}
}

